api/src/main/java/com/graphhopper/routing/util/FlexModel.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util;
+
+import com.graphhopper.util.Helper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FlexModel {
+
+    /**
+     * Converting to seconds is not necessary but makes adding other penalties easier (e.g. turn
+     * costs or traffic light costs etc)
+     */
+    public final static double SPEED_CONV = 3.6;
+    // required
+    private String base;
+    private double maxSpeed;
+    // optional:
+    private double maxPriority = 10;
+    private Double weight, width, height, length;
```

Comment:
These refer to the **maximum** weight/width/height/length a vehicle may have to use a road ? If yes they should be named accordingly. Especially `weight` seems can easily be confused with some edge weight (in the shortest-path weight minimization sense). Maybe call them `maxVehicleWeight/Width/Height/Length` ?

Reply:
The problem I had here is the perspective: if you define the vehicle in the FlexModel then the current approach "allow if flex_model.value<max_value" makes sense. 

But yes, we define more the properties of the road (the vehicle is allowed to drive through) and so we should change this to max_value. Still the Java code then looks wrong "allow if flex_model.max_value<max_value" and max_vehicle_weight was too long for me in the first place ;)

Reply:
Ah you meant specifying the weight here means specifying the weight of the vehicle ?

Reply:
Yeah. That's confusing :D

Reply:
Maybe call it vehicle_weight/width/height/length then (without max) ?

Reply:
Done :)

Reply:
Done :)